src/lib.rs

```diff
@@ -267,8 +296,67 @@ pub async fn main(req: Request, env: Env, _ctx: Context) -> Result<Response> {
+                                                    .into_iter()
+                                                    .map(|p| p.to_string()),
+                                            );
+
+                                            if filter
+                                                .kinds
+                                                .clone()
+                                                .unwrap_or_default()
+                                                .contains(&Kind::WalletConnectRequest)
+                                            {
+                                                let found_events = get_nwc_events(
+                                                    &keys,
+                                                    Kind::WalletConnectRequest,
+                                                    &ctx,
+                                                )
+                                                .await
+                                                .unwrap_or_default();
+                                                events.extend(found_events);
+                                            }
+
+                                            if filter
+                                                .kinds
+                                                .unwrap_or_default()
+                                                .contains(&Kind::WalletConnectResponse)
+                                            {
+                                                let found_events = get_nwc_events(
+                                                    &keys,
+                                                    Kind::WalletConnectResponse,
+                                                    &ctx,
+                                                )
+                                                .await
+                                                .unwrap_or_default();
+                                                events.extend(found_events);
+                                            }
+                                        }
+
+                                        // send all found evens to the user
+                                        for event in events {
+                                            console_log!("sending event to client: {}", &event.id);
+                                            let relay_msg = RelayMessage::new_event(
+                                                subscription_id.clone(),
+                                                event,
+                                            );
+                                            server
+                                                .send_with_str(&relay_msg.as_json())
+                                                .expect("failed to send response");
+                                        }
+
                                         console_log!("ignoring nostr subscription request");
```

Comment:
This log won't always be accurate, maybe change it to "end of subscription request"

```suggestion
                                        console_log!("end of subscription request");
```